### PR TITLE
call back on listen

### PR DESCRIPTION
**Pull request by yoshuawuyts**
makes it so `listen()` calls back when done booting

### CODE REVIEW

**Comment by yoshuawuyts** [comment]
welp, also made it return the server instance so we can listen for events (:

File: index.js [hunk 1]
```diff
@@ -92,18 +92,23 @@ Merry.prototype.start = function () {
   return this._router
 }
 
-Merry.prototype.listen = function (port) {
+Merry.prototype.listen = function (port, cb) {
   assert.ok(this._router, 'merry.listen: router was not found. Did you run app.router() ?')
   var self = this
   var server = http.createServer(this._router)
 
+  cb = cb || noop
```
Comment (lrlna) [inline index.js]: hmmm that's interesting, you just have an blank cb func? I get it, but a lil confusing on first couple of glances i suppose 

a+ on adding a cb though

**Comment by yoshuawuyts** [comment]
What would you propose we do instead?

**Comment by lrlna** [comment]
Well, I am wondering if you think we should assert for a `cb` ? Or do you
think there is a use case for not providing one? Maybe I am just not seeing
something. That way there is no need for noop (it's clever and it works
fine don't get me wrong) and we can just have the assert?

**Pull request merged**